Detectors/Base/include/DetectorsBase/ContVec.h

```diff
@@ -0,0 +1,362 @@
+
+  /// add objects stored in the container to vector<T> branch in the supplied tree
+  void AddToTree(TTree* tree, const std::string& brName);
+  
+ protected:
+
+  /// cast the head of container buffer to internal layout
+  Header*             getHeader() const {return reinterpret_cast<Header*>(getPtr());}
+  T*                  getData()   const {return reinterpret_cast<T*>(getPtr()+dataOffset());}
+
+  /// return next free slot, for "new with placement" creation, not LValue
+  T*                  nextFreeSlot();
+      
+ protected:
+
+  // offset of data wrt buffer start: account for Header + padding after Header block to respect T alignment
+  constexpr sizeType dataOffset() const {return sizeof(Header) + sizeof(Header)%alignof(T);};
+  // for some reason using static constexpr leads to problem in linking the testContVec
+  //  static constexpr sizeType dataOffset() = sizeof(Header) + sizeof(Header)%alignof(T);
+
+  
+  std::unique_ptr<char[]> mPtr;                //! pointer on continuos block containing full object data
+  std::unique_ptr<std::vector<T>> mVecForTree; //! vector for writing objects into the tree
+  
+  ClassDef(ContVec,1)
+};
+
+
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  ContVec<T,H>::ContVec(sizeType iniSize, int expPol) : mPtr(nullptr), mVecForTree(nullptr)
+  {
+    /**
+     * Creates container for objects of 
+     * T POD class
+     * with single description field of H POD class
+     * size: initial capacity of the container
+     * expPol: expansion policy. new_size = expPol>0 ? oldSize+expPol : 2*max(1,oldSize+|expPol|)
+     */
+    static_assert(std::is_trivially_copyable<T>::value,"Object class is not trivially-copiable");
+    static_assert(std::is_trivially_copyable<H>::value,"Header class is not trivially-copiable");
+    reserve(iniSize);
+    if (!expPol) expPol = -1;
+    getHeader()->expandPolicy = expPol;
+  }
+
+//-------------------------------------------------------------------
+template <class T, class H>
+  void ContVec<T,H>::recreate(char* rawPtr, bool copy, sizeType nbytes)
+{
+  /**
+   * recreates container buffer from provided raw pointer 
+   * rawPtr: pointer extracted from the container of simular type using e.g. getPtr() or release() methods
+   * nbytes: expected size of the buffer in bytes, used for error check
+   * copy  : if true, create copy of the array (this is must if the new rawPtr is managed by other object, 
+   *         e.g. was obtained via ContVec::getPtr())
+   *         if false, then the new object will assume ownership over rawPtr
+   */
+  
+  {
+    std::unique_ptr<char[]> dum(mPtr.release()); // destroy old buffer if any (== delete[] mPtr.release())
+  }
+  
+  try {
+    if (rawPtr==nullptr) {
+      throw "invalid arguments:";
+    }
+  } catch(const char* msg) {
+    LOG(FATAL) << msg << " rawPtr is null" << FairLogger::endl;
+  }
```

Comment:
Actually, since it's a container class, IMHO the right thing to do is to let it throw and then catch the exception in the user code.